rero_ils/modules/circ_policies/api.py

```diff
@@ -391,13 +391,25 @@ def can_request(cls, item, **kwargs):
             # can't find any corresponding cipo --> return False
             return False, ["Patron doesn't have the correct role"]
         library_pid = kwargs['library'].pid if kwargs.get('library') \
-            else item.library_pid
+            else record.library_pid
+
+        class_name = type(record).__name__
+        if class_name == 'Item':
+            record_circulation_category_pid = \
+                record.item_type_circulation_category_pid
+        elif class_name == 'Holding':
+            record_circulation_category_pid = \
+                record.circulation_category_pid
+        else:
+            raise Exception(f'Circulation policy: \
+                class is not Holding or Item')
```

Comment:
Message could be more generic : `f'This resource cannot be requested : {record.__class__.__name__}'`